aws-s3/region.ml

```diff
@@ -81,8 +89,16 @@ let of_host host =
   | _ -> failwith "Cannot parse region from host"
 
 let to_host ?(dualstack=false) region =
```

Comment:
I think it would be better is we had a function that would return both host and port.
Something like:
``` 
type endpoint = { inet: [`V4 | V6]; host: string; port: int }
val endpoint: ~inet:[`V4|`V6] -> scheme:[`Http|Https] -> Region.t -> endpoint
```
Possibly in the Region module. Maybe the code would become a bit cleaner, as all the logic of determining the host and port is in the endpoint function.

Reply:
That sounds good.  Then `Http.call` and `Net.connect` can work with `endpoint` values rather than the components that make them up.

I'm trying that now and will send an update to this PR when that's ready.